ndcube/ndcube.py

```diff
@@ -629,6 +630,74 @@ def explode_along_axis(self, axis):
         # Creating a new NDCubeSequence with the result_cubes and common axis as axis
         return NDCubeSequence(result_cubes, meta=self.meta)
 
+    def reproject(self, target_wcs, shape_out=None, order='bilinear', output_array=None, return_footprint=False):
```

Comment:
From sunpy/sunpy#5470 Albert called this `reproject_to` I kind of like that name, thoughts on changing it?